source/MaterialXView/OpenGL/GLUtilityContext.cpp

```diff
@@ -339,22 +328,13 @@ int GLUtilityContext::makeCurrent()
 //
 // Singleton create/destory methods
 //
-GLUtilityContext* GLUtilityContext::create(const WindowWrapper& windowWrapper, HardwareContextHandle context)
+GLUtilityContextPtr GLUtilityContext::creator(const WindowWrapper& windowWrapper, HardwareContextHandle context)
 {
     if (!_globalGLUtilityContext)
     {
         _globalGLUtilityContext = new GLUtilityContext(windowWrapper, context);
     }
-    return _globalGLUtilityContext;
-}
-
-void GLUtilityContext::destroy()
-{
-    if (_globalGLUtilityContext)
-    {
-        delete _globalGLUtilityContext;
-        _globalGLUtilityContext = nullptr;
-    }
+    return std::shared_ptr<GLUtilityContext>(_globalGLUtilityContext);
```

Comment:
This looks a bit risky. If the returned shared pointer is reset or goes out of scope on the receiver side this internal raw pointer will be destroyed, right? It would be safer to store it as a shared pointer internally as well and just return a new copy of that shared pointer.

Reply:
Also, naming the method "creator" is a bit misleading, since that implies that you get a new instance of the class that you get ownership of. That's how the creator pattern is used elsewhere in the code base. And if you call it twice with different inputs you still get the same singleton instance as created by the first call.

It might be better to make the interface something like:

```
void GLUtilityContext::initialize(const WindowWrapper& windowWrapper, HardwareContextHandle context);
GLUtilityContext* GLUtilityContext::instance();
```
to make it more obvious it's a singleton. Since it's a singleton it also feels appropriate that the class itself has ownership of the singleton pointer, so a raw pointer can be returned like above.

On the other hand if it's possible to have multiple GLUtilityContexts in flight at the same time, then using a creator pattern would work, and returning new instances from each creator() call, but then it's not a singleton class anymore.

Reply:
This doesn't need to be a singleton. I'll modify so it actually creates new instances.